dojo/tools/semgrep/parser.py

```diff
@@ -31,6 +31,7 @@ def get_findings(self, filename, test):
                 dynamic_finding=False,
                 vuln_id_from_tool=item["check_id"],
                 nb_occurences=1,
+                unique_id_from_tool=item["extra"]["fingerprint"],
```

Comment:
Might be a better idea to pull this out of the finding instantiation to check if the value is present/valid first. Assigning a potentially empty unique ID could cause all sorts of issues

Reply:
should it be assigned "" or None if no fingerprint actually parsed?

Reply:
Not assigning at all would be ideal. Something like this:

```
             finding = Finding(
                 test=test,
                 title=item["check_id"],
                 severity=self.convert_severity(item["extra"]["severity"]),
                 description=self.get_description(item),
                 file_path=item['path'],
                 line=item["start"]["line"],
                 static_finding=True,
                 dynamic_finding=False,
                 vuln_id_from_tool=item["check_id"],
                 nb_occurences=1,
             )

             unique_id_from_tool = item.get("extra", {}).get("fingerprint")
             if unique_id_from_tool:
                 finding.unique_id_from_tool = unique_id_from_tool
```
This would ensure that the finding would only know about about a unique id if the value is neither a None or an empty string

Reply:
@Maffooch check again, please.

Reply:
all good, approved it.